test/runtests.jl

```diff
@@ -11,8 +11,9 @@ using Statistics: mean
 
 const testdir = dirname(@__FILE__)
 
-testgraphs(g) = [g, Graph{UInt8}(g), Graph{Int16}(g)]
-testdigraphs(g) = [g, DiGraph{UInt8}(g), DiGraph{Int16}(g)]
+testgraphs(g) = is_directed(g) ? [g, DiGraph{UInt8}(g), DiGraph{Int16}(g)] : [g, Graph{UInt8}(g), Graph{Int16}(g)] 
```

Comment:
Maybe it would be easier, if we also made a copy of `g` so that we don't have to make this copy in every loop.

Reply:
not sure I understand. We return `g` in the vector....

Reply:
In the code we have something like
```
 @testset "$g" for g in testgraphs(gx, gdx) 
        @test @inferred(nv(g)) == 15
            gc = copy(g)
            @test add_vertices!(gc, 5) == 5
            @test @inferred(nv(gc)) == 15
        end
```
If we changed that line to
`testgraphs(g) = is_directed(g) ? [copy(g), DiGraph{UInt8}(g), DiGraph{Int16}(g)] : [copy(g), Graph{UInt8}(g), Graph{Int16}(g)]`
then we could omit the line `gc = copy(g)`.

Reply:
I don't think that's right. We use `copy(g)` when we want to modify `g` (to do some checking, for example) within the loop created with`testgraphs`. If we just added a `copy(g)` to `testgraphs`, it wouldn't be a copy of the graph with the changed eltype.